DiskRP Swagger for 09/30

scottlzhang opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

mstomania committed:
add new version

mstomania committed:
add incremental restore point swagger

mstomania committed:
update parameters

mstomania committed:
fix data type

mstomania committed:
fix example parameter name

mstomania committed:
fix example body

mstomania committed:
fix example

mstomania committed:
fix allOf nesting

mstomania committed:
fix property typo

mstomania committed:
fix typo

mstomania committed:
run prettier formatting

mstomania committed:
update readme

mstomania committed:
add default response

mstomania committed:
update version in disk.json

mstomania committed:
address comments

mstomania committed:
fix

mstomania committed:
address comments

mstomania committed:
json beautify

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

Commit:
updated examples and added purchasePlan

yungezz commented:
add waitforArmFeedback since new API version added

yungezz commented:
hi @scottlzhang could you pls fix CI failure by following guideline https://aka.ms/ci-fix

viananth committed:
Add extended location to disks and snapshot

viananth committed:
Add examples to support extended location

viananth committed:
Add new example for disk creation in extended location

viananth committed:
Update edge zone name in examples

rahul602 committed:
Update disk.json

Support to enable disable disk bursting feature would allow customers to burst beyond the provisioned perf target up to 30K IOPS.

chetmaddula committed:
add private endpoint connection ops, examples

scottlzhang committed:
Merge pull request #2 from viananth/vishnu/exl

Add extended location to disks and snapshot

AutorestCI commented:
/azp run automation - sdk

anpodar committed:
DES keyvault and identity changes

anpodar committed:
Fix type

anpodar committed:
Fix spacing

chetmaddula committed:
fix operationIds for private endpoint connections ops

anpodar committed:
Address comments and add example for DES with KV in different sub

scottlzhang committed:
Merge pull request #5 from chetmaddula/master

Add private endpoint connection ops to get, approve, reject, delete for 2020-09-30

rahul602 committed:
Added example for enabled and disable bursting

Fixed review comments.

rahul602 committed:
Updated description for burstingEnabled

meet1993shah committed:
Update disk.json

updated description.

meet1993shah committed:
Update CreateAManagedDiskFromAPlatformImage.json

Update example for creating a managed disk from platform image having purchase plan information

meet1993shah committed:
Update GetInformationAboutAManagedDisk.json

update Get Disk with disk having purchase plan information

meet1993shah committed:
Update ListDiskRestorePointInVmRestorePoint.json

reverting file changes

scottlzhang committed:
Merge pull request #3 from rahul602/master

Update disk.json to add burstingEnabled flag

meet1993shah committed:
Update GetInformationAboutASnapshot.json

updated example for purchase plan

meet1993shah committed:
Update GetInformationAboutASnapshot.json

meet1993shah committed:
Update GetInformationAboutAManagedDisk.json

meet1993shah committed:
Update UpdateAManagedDiskToAddPurchasePlan.json

meet1993shah committed:
Update GetInformationAboutAManagedDisk.json

meet1993shah committed:
Update GetInformationAboutAManagedDisk.json

meet1993shah committed:
Update GetInformationAboutASnapshot.json

meet1993shah committed:
Update CreateAManagedDiskFromAPlatformImage.json

meet1993shah committed:
Update UpdateAManagedDiskToAddPurchasePlan.json

removed isArmResource and faultDomain from the response

scottlzhang committed:
Merge pull request #1 from meet1993shah/dev/likzhan/swagger0930

updated examples and added purchasePlan

scottlzhang committed:
Merge pull request #7 from anpodar/dev/anpodar/DesSwagger

Allow KeyVault resource Id to be optional and allow Identity type None for DiskEncryptionSets

mstomania committed:
Merge branch 'master' of https://github.com/scottlzhang/azure-rest-api-specs into dev/likzhan/swagger0930

scottlzhang committed:
Merge branch 'master' into dev/likzhan/swagger0930

mstomania committed:
Merge branch 'dev/likzhan/swagger0930' of https://github.com/scottlzhang/azure-rest-api-specs into dev/likzhan/swagger0930

mstomania committed:
prettier

mstomania committed:
fix json

mstomania committed:
add missing paramter for DiskAccess

mstomania committed:
fix json examples

chetmaddula commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/ApprovePrivateEndpointConnection.json:
```diff
@@ -0,0 +1,47 @@
+    "privateEndpointConnectionName": "myPrivateEndpointConnection",
+    "privateEndpointConnection": {
+      "properties": {
+        "privateLinkServiceConnectionState ": {
```
please remove space

mstomania committed:
remove extra space

mstomania committed:
add required properties for privateEndPoint example json

mstomania committed:
add missing references for example jsons

mstomania committed:
fix json

mstomania committed:
add missing purchasePlan for DiskUpdateProperties

mstomania committed:
fix readme

mstomania committed:
update purchasePlan property descriptions to match compute.json

mstomania committed:
update purchasePlan's required sequence to match compute.json

mstomania committed:
update purchasePlan description to match compute.json

mstomania committed:
fix privateEndPoint properties

mstomania committed:
fix param for approvePrivateEndpointConnection example

mstomania committed:
change PrivateEndpointConnection to proxyOnlyResource

akning-ms commented:
@ARM reviewer to confirm 
For R4015 - NestedResourcesMustHaveListOperation  whether proxy only Resource is special case?
and R2062 - XmsResourceInPutResponse linter failure

mstomania committed:
add azure-resource for PrivateEndpointConnection

mstomania committed:
remove list diskrestorepoint

mstomania committed:
add description for snapshotsku

pilor commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:
```diff
@@ -0,0 +1,3205 @@
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName}/restorePoints/{vmRestorePointName}/diskRestorePoints/{diskRestorePointName}": {
```
Where are all the APIs related to restorePointCollections and restorePoints?  How does a user discover these deeply nested resources with LISTs and GETs at each level?

scottlzhang replied:
restorePointCollections and restorePoints are created by CRP via privilege calls. The API for createing restorePoints(AllocateRestorePoints) will return the diskRestorePoint armId in response body to client.

mentat9 replied:
What about after creation time? Are there collection gets for restorePointCollections and  restorePointCollections/restorePoints? I didn't find any in other swaggers or this PR. Without those, it seems pretty hard for clients to discover that restorePointCollections contain restorePointCollections/restorePoints (maybe that's somewhat intuitive), and that restorePointCollections/restorePoints contain restorePointCollections/restorePoints/diskRestorePoints.

mentat9 replied:
What about after creation time? Are there collection gets for restorePointCollections and restorePointCollections/restorePoints? I didn't find any in other swaggers or this PR. Without those, it seems pretty hard for clients to discover that restorePointCollections contain restorePointCollections/restorePoints (maybe that's somewhat intuitive), and that restorePointCollections/restorePoints contain restorePointCollections/restorePoints/diskRestorePoints.

mentat9 replied:
What about after creation time? Are there collection gets for restorePointCollections and restorePointCollections/restorePoints? I didn't find any in other swaggers or this PR. Without those, it seems pretty hard for clients to discover that restorePointCollections contain restorePointCollections/restorePoints (maybe that's somewhat intuitive), and that restorePointCollections/restorePoints contain restorePointCollections/restorePoints/diskRestorePoints.

scottlzhang replied:
After creation time, CRP has APIs that return restorePointCollections/restorePoints here link below. 

https://msazure.visualstudio.com/One/_git/Compute-CPlat-Core?path=%2Fsrc%2FCRP%2FDev%2FComputeResourceProvider%2FService%2FAPI%2FControllers%2FRestorePointCollectionsControllerBase.cs&version=GBmaster&line=34&lineEnd=39&lineStartColumn=9&lineEndColumn=79&lineStyle=plain&_a=contents

pilor commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:
```diff
@@ -0,0 +1,3205 @@
+      ],
+      "description": "Properties of disk restore point"
+    },
+    "DiskRestorePointList": {
```
Where is this used?

scottlzhang replied:
This is for LIST operation for DiskRestorePoints. Added LIST operation on line 1603 in disk.json

mstomania committed:
add List diskRestorePoints and List privateEndpointConnections

mstomania committed:
run prettier

mstomania committed:
fix error

scottlzhang committed:
Merge branch 'master' into dev/likzhan/swagger0930

mstomania committed:
fix hypervgeneration'

mstomania committed:
Merge branch 'dev/likzhan/swagger0930' of https://github.com/scottlzhang/azure-rest-api-specs into dev/likzhan/swagger0930

scottlzhang commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:
```diff
@@ -0,0 +1,3205 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/diskAccesses/{diskAccessName}/privateEndpointConnections/{privateEndpointConnectionName}": {
```
Added the LIST operation on line 1513 of disk.json

mentat9 commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:
```diff
@@ -0,0 +1,3303 @@
+      },
+      "description": "A list of private link resources"
+    },
+    "PrivateEndpointConnectionListResult": {
```
You should consider making this schema pageable by annotating it with x-ms-pageable and adding a nextLink property. That way, even if you don't implement it now, you can implement it later without a breaking change.

scottlzhang replied:
Good point. Added pageable in line 1553 for list PrivateEndpointConnection and in line 1649 for list DiskRestorePoints

mstomania committed:
add pageable for list privateEndPointConnections and list diskRestorePoints

mstomania committed:
add missing nextLink for ListPrivateEndpointConnectionsResult

mstomania committed:
prettier

mstomania committed:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs into dev/likzhan/swagger0930

heaths reviewed (commented):
Key Vault reformatting looks fine. We have a PR out that should fix that for everyone that should be merged soon. /cc @herveyw-msft @Lusitanian (I keep seeing these same three files reformatted over and over again - good reason to get that PR in sooner than later 😉).

yungezz commented:
hi @msyyc could you pls help to confirm impact to Python SDK track1 and track2? thanks.

yungezz commented:
hi @ArcturusZhang could you pls help to look at the go SDK generation failure? thanks

yungezz commented:
> hi @ArcturusZhang could you pls help to look at the go SDK generation failure? thanks

For GO SDK failure, it's caused by readme validation `the output-folder of a management plane package 'profile-hybrid-2020-09-01' must be in this pattern: '^services(/preview)?/[^/]+/mgmt/[^/]+/[^/]+$'`. Seems not caused by this PR. But need to be fixed. @scottlzhang

ArcturusZhang commented:
/azp run

Sandido commented:
@yungezz , this PR re-introduced the preview version of the Gallery API, 2020-09-30. 
This version [had been reverted](https://github.com/Azure/azure-rest-api-specs/pull/12222/files#diff-5871f1ff4953a3d11f8779e01460fe0ae3cb27630c3868c55a53417859e83a0aL222) and should not have been re-introduced.
